File: app/src/main/java/com/example/hyp/meizhi/MeizhiAppliaction.java
package com.example.hyp.meizhi;

import android.app.Application;

import com.example.hyp.net.RetrofitManager;

public class MeizhiAppliaction extends Application {

    private static String BaseUrl = "http://gank.io/";
    private static String cacheDir = "meizhi";
    private static String cacheName = "mei-zhi";
    private static long size = 1024 * 1024 * 100;

    @Override
    public void onCreate() {
        super.onCreate();

        initRetrofit();
    }

    private void initRetrofit() {
        RetrofitManager.INSTANCE.initRetrofit(this)
                .baseUrl(BaseUrl)
                .cacheDir(cacheDir)
                .cacheName(cacheName)
                .cacheSize(size)
                .readTimeout(500)
                .build();
    }
}

File: app/src/main/java/com/example/hyp/meizhi/MeizhiAppliaction.kt
package com.example.hyp.meizhi

import android.app.Application

import com.example.hyp.net.RetrofitManager

class MeizhiAppliaction : Application() {

    override fun onCreate() {
        super.onCreate()

        initRetrofit()
    }

    private fun initRetrofit() {
        RetrofitManager.initRetrofit(this)
                .baseUrl(BaseUrl)
                .cacheDir("meizhi")
                .cacheName(cacheName)
                .cacheSize(size)
                .readTimeout(500)
                .build()
    }

    companion object {

        private val BaseUrl = "http://gank.io/"
        private val cacheName = "mei-zhi"
        private val size = (1024 * 1024 * 100).toLong()
    }
}
